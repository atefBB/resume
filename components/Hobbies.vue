<template>
  <div v-if="hobbies.hobbies_content" class="container">
    <h3 id="hobbies-title">
      {{
        hobbies.hobbies_title
          ? hobbies.hobbies_title
          : "Activities and interests"
      }}
    </h3>
    <el-card id="hobbies-content">
      {{ hobbies.hobbies_content }}
    </el-card>
  </div>
</template>

<script>
import hobbies from "~/assets/yaml/hobbies.yml"
import { safeLoad } from "js-yaml"

export default {
  props: [],
  data() {
    return {
      hobbies: {}
    }
  },
  created: function() {
    this.getContent()
  },
  methods: {
    getContent() {
      this.hobbies = safeLoad(hobbies)
    }
  }
}
</script>
