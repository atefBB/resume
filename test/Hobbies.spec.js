import faker from "faker"
import { testHelper, mountHelper } from "./helpers"
import Hobbies from "~/components/Hobbies.vue"
jest.mock(`~/assets/yaml/hobbies.yml`)

describe("Hobbies", () => {
  testHelper(Hobbies)

  describe("Matches randoms values", () => {
    const values = {
      hobbies: {
        hobbies_title: faker.hacker.adjective(),
        hobbies_content: faker.lorem.paragraph()
      }
    }
    const mockFn = jest.fn()
    const options = {
      methods: {
        getContent: mockFn
      }
    }
    const factory = mountHelper(Hobbies)

    it("Should call the mocked function", () => {
      factory({ options, values })
      expect(mockFn).toHaveBeenCalled() //we use correctly the mock function
    })

    //hobbies_title
    it("Should contain the hobbies_title", () => {
      const wrapper = factory({ options, values })
      expect(wrapper.find("#hobbies-title").text()).toEqual(
        expect.stringContaining(values.hobbies.hobbies_title)
      )
    })
    it("Should contain the default hobbies_title", () => {
      let localValues = {
        hobbies: { ...values.hobbies, hobbies_title: undefined }
      }
      const wrapper = factory({ options, values: localValues })
      expect(wrapper.find("#hobbies-title").text()).toEqual(
        expect.stringContaining("Activities and interests")
      )
    })

    //hobbies_content
    it("Should contain the hobbies_content", () => {
      const wrapper = factory({ options, values })
      expect(wrapper.find("#hobbies-content").text()).toEqual(
        expect.stringContaining(values.hobbies.hobbies_content)
      )
    })
    it("Should NOT contain the hobbies section", () => {
      let localValues = {
        hobbies: { ...values.hobbies, hobbies_content: undefined }
      }
      const wrapper = factory({ options, values: localValues })
      expect(wrapper.isEmpty()).toBe(true)
    })
  })
})
