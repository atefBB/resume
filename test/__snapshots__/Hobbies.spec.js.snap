// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hobbies Default Template 1`] = `
<div class="container">
  <h3 id="hobbies-title">
    Activities and interests
  </h3>
  <div class="el-card is-always-shadow" id="hobbies-content">
    <!---->
    <div class="el-card__body">
      Sports (cycling, running, triathlon etc.), High Tech actualities (IOT, CTF etc.), epicurean, earth and environment, ‘Mens sana in corpore sano’

    </div>
  </div>
</div>
`;
